bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/discovery/RadioThermostatDiscoveryService.java

```diff
@@ -0,0 +1,301 @@
+     * @return
+     * @throws IOException
+     */
+
+    protected void parseResponse(String response) {
+        DiscoveryResult result;
+
+        String name = "unknownName";
+        String uuid = "unknownThermostat";
+        String ip = null;
+        String url = null;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.debug("key: {} value: {}.", key, value);
+            switch (key) {
+                case "location":
+                    try {
+                        url = value;
+                        ip = new URL(value).getHost();
+                    } catch (MalformedURLException e) {
+                        logger.debug("Malfored URL {}", e.getMessage());
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+        scanner.close();
+
+        logger.debug("Found thermostat, ip: {} ", ip);
+
+        if (ip == null) {
+            logger.debug("Bad Format from thermostat");
+            return;
+        }
+
+        JsonObject content;
+        String sysinfo;
+
+        try {
+            // Run the HTTP request and get the JSON response from the thermostat
+            ContentResponse contentResponse = httpClient.newRequest(url).method(GET).timeout(20, TimeUnit.SECONDS)
+                    .send();
+            sysinfo = contentResponse.getContentAsString();
+            content = new JsonParser().parse(sysinfo).getAsJsonObject();
+            uuid = content.get("uuid").getAsString();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.debug("Cannot get system info from thermostat {} {}", ip, e.getMessage());
+            sysinfo = null;
+        }
+
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(url + "name").method(GET)
+                    .timeout(20, TimeUnit.SECONDS).send();
+            String nameinfo = contentResponse.getContentAsString();
+            content = new JsonParser().parse(nameinfo).getAsJsonObject();
+            name = content.get("name").getAsString();
+        } catch (Exception e) {
+            logger.debug("Cannot get name from thermostat {} {}", ip, e.getMessage());
+        }
+
+        logger.debug("Discovery returned: {} uuid {} name {}", sysinfo, uuid, name);
+
+        ThingUID thingUid = new ThingUID(RadioThermostatBindingConstants.THING_TYPE_RTHERM, uuid);
+
+        logger.debug("Got discovered device.");
+
+        String label = String.format("RadioThermostat (%s)", name);
+        result = DiscoveryResultBuilder.create(thingUid).withLabel(label).withRepresentationProperty(uuid)
```

Comment:
I assume it is supposed to be this:
```suggestion
        result = DiscoveryResultBuilder.create(thingUid).withLabel(label)
                      .withRepresentationProperty(RadioThermostatBindingConstants.PROPERTY_IP)
```

Reply:
fixed